packages/kbn-apm-synthtrace/src/lib/apm/aggregators/service_metrics_aggregator.ts

```diff
@@ -141,22 +139,24 @@ export class ServicMetricsAggregator implements StreamAggregator<ApmFields> {
       }
 
       const state = this.state[key];
-      state.count++;
-
-      switch (event['event.outcome']) {
-        case 'failure':
-          state.failure_count++;
-          break;
-        case 'success':
-          state.success_count++;
-          break;
-      }
 
       const duration = Number(event['transaction.duration.us']);
+
       if (duration >= 0) {
+        state.count++;
+
         state.sum += duration;
         if (duration > state.max) state.max = duration;
         if (duration < state.min) state.min = Math.min(0, duration);
+
+        switch (event['event.outcome']) {
+          case 'failure':
+            state.failure_count++;
+            break;
+          case 'success':
+            state.success_count++;
+            break;
+        }
```

Comment:
I figured it out that in some cases duration is undefined. In that case we don't want to increase the counters that's why I moved them inside the if statement